build-tools/src/main/kotlin/org/jetbrains/kotlin/KLibInstall.kt

```diff
@@ -12,17 +13,16 @@ import java.io.File
 
 // TODO: Implement as a part of the gradle plugin
 open class KlibInstall: Exec() {
-    @InputFile
-    lateinit var klib: File
+    @Input
+    lateinit var klib: Provider<File>
 
     @Input
     var repo: File = project.rootDir
 
-    val installDir: File
-        @OutputDirectory
-        get() {
-            val klibName = klib.name.take(klib.name.lastIndexOf('.'))
-            return project.file("${repo.absolutePath}/$klibName")
+    val installDir: Provider<File>
```

Comment:
May be @Output required?

Reply:
Only `@OutputDirectory` exists in gradle

Reply:
But it's still an output directory and we still need to tell about it to Gradle.
`@OutputDirectory` works well with providers too, so I think we can just keep this annotation here.